src/main/java/com/amazon/randomcutforest/serialize/RandomCutForestSerDe.java

```diff
@@ -77,6 +87,29 @@ public String toJson(RandomCutForest rcf) {
         return gson.toJson(rcf);
     }
 
+    /**
+     * Serializes a RCF object to ion binary.
+     *
+     * @param rcf a RCF object
+     * @return ion binary serialized from the RCF
+     * @throws IonException when the input cannot be serialized
+     */
+    public byte[] toIon(RandomCutForest rcf) {
+        try {
+            IonReaderBuilder readerBuilder = IonReaderBuilder.standard();
+            IonReader reader = readerBuilder.build(toJson(rcf));
```

Comment:
Ion supports more types than JSON. Are we losing out by going through the JSON serializer first? If we're not taking advantage of other Ion features, is this actually more efficient (in terms of compute time or storage space) than just gzipping the JSON output?

Reply:
The primary reason of reusing existing json is it's ready as building block. I tried IonMapper, customization of serialization/deserialization for non-pojo objects with many final private fields proves a challenge, which Gson works around nicely.

For our use case, serialization cost is infrequent but storage cost is long-lived. I thought about gzipping but am concerned the savings are not consistent or optimal. It is however worth an experiment.

Reply:
This pr is more of a test to see if this use case is specific-purpose.

Reply:
I see. I think the only concern would be if at some point we want to create a "proper" Ion format, I think it would be incompatible with this change (we would either end up with two `toIon` methods, or we'd have a backwards-incompatible change to `toIon`/`fromIon`. How do you think we should handle that?